service/discovery/azure/compute.go

```diff
@@ -724,9 +618,179 @@ func (d *azureComputeDiscovery) keyURL(diskEncryptionSetID string) (string, *arm
+	}
+
+	// Check if resource is public available
+	if util.Deref(site.Properties.PublicNetworkAccess) == "Enabled" {
+		publicAccess = true
+	}
+
+	// Get resource logging status
+	resourceLogging = d.getResourceLoggingWebApp(site)
+
+	return ni, publicAccess, resourceLogging
+}
+
+// getResourceLoggingWebApp determines if logging is activated for given web app by checking the respective app setting
+func (d *azureComputeDiscovery) getResourceLoggingWebApp(site *armappservice.Site) (rl *voc.ResourceLogging) {
+	rl = &voc.ResourceLogging{Logging: &voc.Logging{}}
+
+	if site == nil {
+		log.Error("given parameter is empty")
+		return
+	}
+
+	appSettings, err := d.clients.webAppClient.ListApplicationSettings(context.Background(),
+		*site.Properties.ResourceGroup, *site.Name, &armappservice.WebAppsClientListApplicationSettingsOptions{})
+	if err != nil {
+		log.Errorf("could not get application settings for '%s': %v", util.Deref(site.Name), err)
+		return
+	}
+	if appSettings.Properties["APPLICATIONINSIGHTS_CONNECTION_STRING"] != nil {
+		rl.Enabled = true
+		// TODO: Get id of logging service and add it (currently not possible via app settings): rl.LoggingService
+
+	}
+	return
+
+}
+
+// getRedundancy returns the redundancy status
+func getRedundancy(app *armappservice.Site) *voc.Redundancy {
+	r := &voc.Redundancy{}
+	switch util.Deref(app.Properties.RedundancyMode) {
+	case armappservice.RedundancyModeNone:
+		break
+	case armappservice.RedundancyModeActiveActive:
+		r.Zone = true
+	case armappservice.RedundancyModeFailover, armappservice.RedundancyModeGeoRedundant:
+		r.Zone = true
+		r.Geo = true
+	}
+	return r
+}
+
+// We really need both parameters since config is indeed more precise but it does not include the `httpsOnly` property
+func getTransportEncryption(siteProperties *armappservice.SiteProperties, config armappservice.WebAppsClientGetConfigurationResponse) (enc *voc.TransportEncryption) {
+	var (
+		tlsVersion string
+	)
+
+	// Check TLS version
+	switch util.Deref(config.Properties.MinTLSVersion) {
+	case armappservice.SupportedTLSVersionsOne2:
+		tlsVersion = constants.TLS1_2
+	case armappservice.SupportedTLSVersionsOne1:
+		tlsVersion = constants.TLS1_1
+	case armappservice.SupportedTLSVersionsOne0:
+		tlsVersion = constants.TLS1_0
+	}
+
+	// Create transportEncryption voc object
+	if tlsVersion != "" {
+		enc = &voc.TransportEncryption{
+			Enforced:   util.Deref(siteProperties.HTTPSOnly),
+			TlsVersion: tlsVersion,
+			Algorithm:  string(util.Deref(config.Properties.MinTLSCipherSuite)),
+			Enabled:    true,
+		}
+	} else {
+		enc = &voc.TransportEncryption{
+			Enforced:  util.Deref(siteProperties.HTTPSOnly),
+			Enabled:   false,
+			Algorithm: string(util.Deref(config.Properties.MinTLSCipherSuite)),
+		}
+	}
+
+	return
+}
+
+// runtimeInfo returns the runtime language and version
+func runtimeInfo(runtime string) (runtimeLanguage string, runtimeVersion string) {
+	if runtime == "" || !strings.Contains(runtime, "|") {
```

Comment:
Maybe add a lil bit of explanation why "|" is so important here.

Reply:
I don't think we need to explain that, it's a normal string from Azure and they use the pipe and if we don't check that, we get a string split error just like any other string splits.